packages/astro/src/runtime/server/index.ts

```diff
@@ -278,8 +278,9 @@ If you're still stuck, please open an issue on GitHub or join us at https://astr
 		return markHTMLString(html.replace(/\<\/?astro-fragment\>/g, ''));
 	}
 
-	// Include componentExport name and componentUrl in hash to dedupe identical islands
-	const astroId = shorthash.unique(`<!--${metadata.componentExport!.value}:${metadata.componentUrl}-->\n${html}`);
+	// Include componentExport name, componentUrl, and props in hash to dedupe identical islands
+	const stringifiedProps = JSON.stringify(props);
```

Comment:
Note: this _will not_ solve the issue for non-JSON props! When passing a function as a prop for instance:

```astro
---
import Counter from '../components/Counter.svelte';

function test() {
	return 'nice!'
}
---
<Counter client:visible nice="cool" testFn={test}>

```

...`stringifiedProps` will have this value:
```js
'{"nice":"cool","class":"astro-3GOAJIZA"}'
// note testFn is missing
```

Doesn't look like `JSON.stringify` will throw on unrecognized keys. Just want to call out this doesn't fix 100% of use cases

Reply:
@bholmesdev Fixed in 63a1939. 

> We don't _really_ support passing functions to components since they might reference something in a parent scope or have other dependencies... But you're right that `JSON.stringify` didn't support everything that could be a prop.

Reply:
Huh, never knew about `serializeProps`. This is great!